### DISCUSSION TITLE
@apply won't be parsed

### DISCUSSION THREAD

**Discussion by BaldFabi** [tailwindlabs/tailwindcss#11850] [Help] [answered]
I'm currently trying to use tailwind in combination with a static site (written by hand - yes it's on purpose for now).

I've created a folder and ran `npm init`, `npm i -D tailwindcss autoprefixer postcss` and `npx tailwindcss init` in it.
After that I created/updated the `tailwind.config.js` and `postcss.config.js`.
`tailwind.config.js`
```javascript
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./templates/**/*.{html,css}"],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

`postcss.config.js`
```javascript
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};
```

As you may already see due to the `tailwind.config.js` file I also created a `templates` folder.
This folder contains a `index.html` and `main.css` file.

`index.html`
```html
<html>
  <head>
    <link href="styles.css" rel="stylesheet" />
  </head>
  <body>
    <h1 class="text-xl">test</h1>
    <p class="bg-red-50">hello</p>
    <h2>foobar</h2>
  </body>
</html>
```

`main.css`
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

h2 {
  @apply text-4xl;
}
```

After running `npx tailwindcss --postcss postcss.config.js -o templates/styles.css` the `styles.css` file will be created and all necessary classes are present except for the class(es) in the `@apply` directive in the `main.css` file.
I actually don't know if the file isn't even parsed or the `@apply` directive isn't working.
It also makes no difference if the `main.css` file is linked in the html.

Am I missing something? I hope someone has a clue to help out

**Comment by wongjn** [comment, marked as answer]
You are indeed missing something. You'd want to provide `main.css` as an input stylesheet to Tailwind via the `-i`/`--input` parameter:

```sh
npx tailwindcss --input templates/main.css --postcss postcss.config.js -o templates/styles.css
```

Also, as an aside, consider not having the `css` extension in your `content` file globs, as per [the documentation](https://tailwindcss.com/docs/content-configuration#pattern-recommendations):

> It’s also important that you don’t scan any CSS files — configure Tailwind to scan your templates where your class names are being used, never the CSS file that Tailwind is generating.
>
> **Never include CSS files in your content configuration**

**Reply by wongjn** [reply]
> I actually expected that having `.css` in the tailwind config would be enough. I've removed `.css` now from the config file and used the `-i` parameter but unfortunately it doesn't change anything. In the `styles.css` is still only the `.text-xl` and `.bg-red-50` selector but not the `.text-4xl` (additionally to the tailwind default selectors).[^1]

`@apply` would not create a `.text-4xl` CSS rule – `@apply` adds CSS properties from the classes supplied to it into the rule it resides in. So, you should see a CSS rule like:

```css
h2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
}
```

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/11850#discussioncomment-6763859

**Reply by BaldFabi** [reply]
You're totally right. Obviously I didn't used my brain after I tried the changes, thank you

**Comment by BaldFabi** [comment]
I actually expected that having `.css` in the tailwind config would be enough. I've removed `.css` now from the config file and used the `-i` parameter but unfortunately it doesn't change anything.
In the `styles.css` is still only the `.text-xl` and `.bg-red-50` selector but not the `.text-4xl` (additionally to the tailwind default selectors).

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/11850#discussioncomment-6763890 to keep the reply chain together.
